### PR TITLE
[Merged by Bors] - feat: trigger automated Zulip emojis via `bors x` command

### PR DESCRIPTION
**Pull request by adomani**
Changes the way in which emoji reactions to `delegated`, `ready-to-merge` and `merged` are applied in Zulip: instead of running periodically a script to add the reactions, the reactions are now applied at the same time that the label is applied, or when PRs are merged into master.

---

The test PR is #19367: I was using #19367 as PR to test, but the mechanism to apply/remove the labels was simply using `push`, since triggering the action with a comment could only be done if the script was already on master.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — COMMENTED** [review]
Instead of the current "narrow" in the zulip search, I think we should search for messages that include the PR_NUMBER. That will be a much more efficient zulip search.

**Comment by adomani** [comment]
I removed the `narrow` restriction, but I could not find a way to filter directly with the Zulip python API all the messages that contain a given PR number.  Do you know if it is possible/how to do it?

File: scripts/zulip_emoji_merge_delegate.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
 #!/usr/bin/env python3
 import sys
 import zulip
-import requests
 import re
 
 # Usage:
-# python scripts/zulip_emoji_merge_delegate.py $ZULIP_API_KEY $ZULIP_EMAIL $ZULIP_SITE $GITHUB_TOKEN
+# python scripts/zulip_emoji_merge_delegate.py $ZULIP_API_KEY $ZULIP_EMAIL $ZULIP_SITE $LABEL $PR_NUMBER
```
Comment (jcommelin) [inline scripts/zulip_emoji_merge_delegate.py:7]: Please also update `scripts/README.md`.

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by adomani** [comment]
bors merge

**Pull request closed without merging**